runtime/v2/shim/util.go

```diff
@@ -116,41 +117,60 @@ func Connect(address string, d func(string, time.Duration) (net.Conn, error)) (n
 }
 
 // WritePidFile writes a pid file atomically
-func WritePidFile(path string, pid int) error {
-	path, err := filepath.Abs(path)
+func WritePidFile(path string, pid int) (err error) {
+	path, err = filepath.Abs(path)
 	if err != nil {
 		return err
 	}
-	tempPath := filepath.Join(filepath.Dir(path), fmt.Sprintf(".%s", filepath.Base(path)))
-	f, err := os.OpenFile(tempPath, os.O_RDWR|os.O_CREATE|os.O_EXCL|os.O_SYNC, 0666)
+
+	f, err := os.CreateTemp(filepath.Dir(path), "."+filepath.Base(path))
```

Comment:
We lose O_SYNC (for both functions) with this change. https://cs.opensource.google/go/go/+/refs/tags/go1.20.1:src/os/tempfile.go;l=44

Reply:
use `f.Sync()` resolve